sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -1196,13 +1196,16 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
 
     selectQuery match {
       case Some(q) =>
-        // Hive does not allow to use a CTAS statement to create a partitioned table.
+        // Hive does not allow to use a CTAS statement to create a partitioned table
+        // by specifying table schema.
         if (tableDesc.partitionColumnNames.nonEmpty) {
           val errorMessage = "A Create Table As Select (CTAS) statement is not allowed to " +
-            "create a partitioned table using Hive's file formats. " +
+            "create a partitioned table using Hive's file formats by specifying table schema. " +
             "Please use the syntax of \"CREATE TABLE tableName USING dataSource " +
             "OPTIONS (...) PARTITIONED BY ...\" to create a partitioned table through a " +
-            "CTAS statement."
+            "CTAS statement. Since Spark 3.0, you can specify partition column names " +
```

Comment:
we don't need to put spark version in error message. Users can only see this message when they use Spark 3.0.

Reply:
Ok. Removed version info.